update python (#19)
hsf-training/analysis-essentials: python/methods.md

Generally, you should only use `lambda` methods to define little throw-away
methods. The main downside with using them is that you can’t attach a docstring
to them, and they become unwieldy when there’s complex logic.

Golden rules:

>  - make a function idempotent if possible (stateless, the same input values will return the same output). This is of course different for classes.

Reviewer: ```suggestion
 - Make functions idempotent where possible (stateless, the same input values will return the same output). This is of course different for classes.
```
